components/email_aliases/email_aliases.mojom

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+module email_aliases.mojom;
+
+struct Alias {
+  string email;
+  string? note;
+  array<string>? domains;
+};
+
+// Observer for change in email aliases state
+interface EmailAliasesServiceObserver {
+
+  // Called when aliases have been updated.
+  OnAliasesUpdated(array<Alias> aliases);
+
+  // Called when user logs in to an account.
+  OnLoggedIn(string primaryEmail);
```

Comment:
If we're already logged in and open a new settings page, how do know the current login state?

Reply:
My thinking was that, on the C++ side, `AddObserver` would call `OnLoggedIn`.

Reply:
I hope I have clarified this via https://github.com/brave/brave-core/pull/28543#discussion_r2089634549